### ISSUE TITLE
Enable `clearMocks`, `resetMocks`, and `restoreMocks` by default

### ISSUE THREAD

**Issue by JamieMagee** [renovatebot/renovate#18768]
### Describe the proposed change(s).

Jest config allows you to specify 3 flags regarding mocks in increasing order of strictness:

1. `clearMocks: true` clears the call count and args of any mocks before each test.
2. `resetMocks: true` goes a step further, and also clears any mock implementations / return values defined before each test.
3. `restoreMocks: true` is the strictest and actually removes all mocks from all functions before each test.

The default, unfortunately, is not to do any of this, which means out-of-the-box Jest makes it very easy to write unit tests that unintentionally rely on test ordering.

I think the bare minimum should be to enable `clearMocks` in [`jest.config.ts`][1] (and remove any [explicit `jest.clearAllMocks()` calls][2]). Ideally I would also like to enable `resetMocks` (and remove any [explicit `jest.resetAllMocks()` calls][3]). I'd like to evaluate use of `restoreMocks` once `resetMocks` is reached.

See https://github.com/facebook/jest/issues/10242 for more context

[1]: https://github.com/renovatebot/renovate/blob/main/jest.config.ts
[2]: https://github.com/search?q=repo%3Arenovatebot%2Frenovate%20clearAllMocks&type=code
[3]: https://github.com/search?q=repo%3Arenovatebot%2Frenovate%20resetAllMocks&type=code

**Comment by viceice** [comment]
I'm ok with 1 but don't like 2 and 3. 2 and 3 are onyl for lazy people whow don't want to think about test order and side effects.
2 and 3 can also cause another performance penality and makes tests even slower than they already are.

**Comment by JamieMagee** [comment]
I agree about the potential for performance penalties, but I think it's a trade-off between test correctness and performance. Already, enabling `clearMocks`, I see test failures due to tests unintentionally relying on setup from other tests.

**Comment by HonkingGoose** [comment]
Looks like PR #20402 fixes something about mocking as well:

> ## Context
>
> - `jest.resetAllMocks()` now resets spy's, so don't reset if nor required
> - if a spy has no mock implementation, it calls the original
